volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,165 @@
+                                                  layer_name = layer.name)
+
+                # There is no field that has a count of Attributes
+                # Keep Attempting to read attributes until we get an invalid attr_header.AttrType
+                while attr_header.AttrType in attr_types.choices.values():
+                    vollog.debug(f"Attr Type: {attr_header.AttrType}")
+
+                    # Offset past the headers to the attribute data
+                    attr_data_offset = offset + attr_base_offset + self.context.symbol_space.get_type(
+                        attribute_object).relative_child_offset("Attr_Data")
+
+                    # MFT Flags determine the file type or dir
+                    if mft_record.Flags in mft_flags.choices.values():
+                        mft_flag = mft_flags.lookup(mft_record.Flags)
+                    else:
+                        mft_flag = hex(mft_record.Flags)
+
+                    # Standard Information Attribute
+                    if attr_header.AttrType == attr_types.choices.get('STANDARD_INFORMATION'):
+                        attr_data = self.context.object(si_object, offset = attr_data_offset, layer_name = layer.name)
+
+                        yield 0, (
+                            format_hints.Hex(attr_data_offset),
+                            mft_record.get_signature(),
+                            mft_record.RecordNumber,
+                            mft_record.LinkCount,
+                            mft_flag,
+                            renderers.NotApplicableValue(),
+                            attr_types.lookup(attr_header.AttrType),
+                            conversion.wintime_to_datetime(attr_data.CreationTime),
+                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
+                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
+                            conversion.wintime_to_datetime(attr_data.AccessedTime),
+                            renderers.NotApplicableValue(),
+                        )
+
+                    # File Name Attribute
+                    if attr_header.AttrType == attr_types.choices.get('FILE_NAME'):
+                        attr_data = self.context.object(fn_object, offset = attr_data_offset, layer_name = layer.name)
+                        file_name = attr_data.get_full_name()
+                        if attr_data.Flags in permission_flags.choices.values():
+                            permissions = permission_flags.lookup(attr_data.Flags)
+                        else:
+                            permissions = hex(attr_data.Flags)
+
+                        yield 1, (format_hints.Hex(attr_data_offset), mft_record.get_signature(),
+                                  mft_record.RecordNumber, mft_record.LinkCount, mft_flag, permissions,
+                                  attr_types.lookup(attr_header.AttrType),
```

Comment:
This also could just be `attr_header.AttrType.lookup()`.  The Enum already knows what type it is, so the object carries around all its possible choices.